### ISSUE TITLE
Bug: (sort-classes) Overloads are not properly sorted.

### ISSUE THREAD

**Issue by ElPrudi** [azat-io/eslint-plugin-perfectionist#46]
### Describe the bug

It's seems like there is an error in the way you handle overloads. The top most overload is always treated as a separate function and because it has no body, TypeScript doesn't like this.

### Code example

For example, it turns this:

```ts
class X {
    setBackground(color: number, hexFlag: boolean): this
    setBackground(color: Color | string | CSSColor): this
    setBackground(r: number, g: number, b: number, a?: number): this
    setBackground(color: ColorArgument, arg1?: boolean | number, arg2?: number, arg3?: number): this {
        <body>
    }
}
```

into this:

```ts
class X {
    setBackground(r: number, g: number, b: number, a?: number): this
    setBackground(color: CSSColor | Color | string): this
    setBackground(color: ColorArgument, arg1?: boolean | number, arg2?: number, arg3?: number): this {
        <body>
    }
    setBackground(color: number, hexFlag: boolean): this
}
```

### ESLint version

8.45.0

### ESLint Plugin Perfectionist version

1.5.0

### Validations

- [X] Read the [docs](https://eslint-plugin-perfectionist.azat.io).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by azat-io** [comment]
Thanks for the issue! I'll fix this problem soon.

**Comment by azat-io** [comment]
Fixed in c4939b44de0ea9e77a98cee76d4043926d3a02a8.

Release coming soon.

**Comment by azat-io** [comment]
Released in [v2.0.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v2.0.0)
